backends/bmv2/CMakeLists.txt

```diff
@@ -66,18 +67,18 @@ set (BMV2_BACKEND_HDRS
   )
 
 set (IR_DEF_FILES ${IR_DEF_FILES} ${CMAKE_CURRENT_SOURCE_DIR}/bmv2.def PARENT_SCOPE)
+add_cpplint_files (${CMAKE_CURRENT_SOURCE_DIR} "${BMV2_BACKEND_SRCS}")
 
+build_unified(BMV2_BACKEND_SRCS)
 add_library(bmv2backend ${BMV2_BACKEND_SRCS})
 add_dependencies(bmv2backend genIR)
 
+build_unified(BMV2_SRCS)
 add_executable(p4c-bm2-ss ${BMV2_SRCS})
 target_link_libraries (p4c-bm2-ss bmv2backend ${P4C_LIBRARIES} ${P4C_LIB_DEPS})
 
 install(TARGETS p4c-bm2-ss RUNTIME DESTINATION ${P4C_RUNTIME_OUTPUT_DIRECTORY})
 
-set (BMV2_SOURCES ${BMV2_BACKEND_SRCS} ${BMV2_SRCS} ${BMV2_BACKEND_HDRS})
-add_cpplint_files (${CMAKE_CURRENT_SOURCE_DIR} "${BMV2_SOURCES}")
-
 # hack to get around the fact that the test scripts expect the backend
```

Comment:
Does the comment below still apply?

Reply:
Nothing has really changed for code outside this file yet, so I believe that comment still applies (though I'll look over it again when I'm not on my phone). I plan to rework the GTest extension stuff in a followup PR, so expect some cleanup of this file soon.

Reply:
Oh, I see. I thought this was in reference to a different comment in `test/CMakeLists.txt`.

I had the same question when I was looking at this file. I would expect this to be obsolete, since after all we've long since remove automake. @cc10512, can we remove this hack?

Reply:
We did remove automake, but the runscripts are still hardcoded for the build/p4c path. Ideally we'll change the runscripts to invoke p4c (the driver), and then all these can be removed. That and how we handle includes. Again, p4c does solve that issue as well.

Reply:
OK. File it under "TODO", then.